src/logion/controllers/locrequest.controller.ts

```diff
@@ -0,0 +1,90 @@
+import { injectable } from "inversify";
+import { Controller, ApiController, HttpPost, Async, HttpPut } from "dinoloop";
+import { components } from "./components";
+import { v4 as uuid } from "uuid";
+import moment from "moment";
+import { LocRequestStatus } from "../model/locrequest.model";
+import { ALICE, BOB } from "../model/addresses.model";
+import { OpenAPIV3 } from "express-oas-generator";
+import { getRequestBody, getDefaultResponses, addTag, setControllerTag } from "./doc";
+
+export function fillInSpec(spec: OpenAPIV3.Document): void {
+    const tagName = 'LOC Requests';
+    addTag(spec, {
+        name: tagName,
+        description: "Handling of LOC Requests"
+    });
+    setControllerTag(spec, /^\/api\/loc-request.*/, tagName);
+
+    LocRequestController.createLocRequest(spec);
+    LocRequestController.fetchRequests(spec);
+}
+
+type CreateLocRequestView = components["schemas"]["CreateLocRequestView"];
+type LocRequestView = components["schemas"]["LocRequestView"];
+type FetchLocRequestsSpecificationView = components["schemas"]["FetchLocRequestsSpecificationView"];
+type FetchLocRequestsResponseView = components["schemas"]["FetchLocRequestsResponseView"];
+
+@injectable()
+@Controller('/loc-request')
+export class LocRequestController extends ApiController {
+
+    static createLocRequest(spec: OpenAPIV3.Document) {
+        const operationObject = spec.paths["/api/loc-request"].post!;
+        operationObject.summary = "Creates a new LOC Request";
+        operationObject.description = "The authenticated user must be the requester. No signature required";
+        operationObject.requestBody = getRequestBody({
+            description: "LOC Request creation data",
+            view: "CreateLocRequestView",
+        });
+        operationObject.responses = getDefaultResponses("LocRequestView");
+    }
+
+    @HttpPost('')
+    @Async()
+    async createLocRequest(createLocRequestView: CreateLocRequestView): Promise<LocRequestView> {
+        return this.fake(createLocRequestView, "PENDING");
+    }
+
+    private fake(createLocRequestView: CreateLocRequestView, status: LocRequestStatus): LocRequestView {
+        return {
+            description: createLocRequestView.description,
+            requesterAddress: createLocRequestView.requesterAddress,
+            ownerAddress: createLocRequestView.ownerAddress,
+            id: uuid(),
+            createdOn: moment().toISOString(),
+            status
+        }
+    }
+
+    static fetchRequests(spec: OpenAPIV3.Document) {
+        const operationObject = spec.paths["/api/loc-request"].put!;
+        operationObject.summary = "Lists LOC Requests based on a given specification";
+        operationObject.description = "The authenticated user must be either expected requester or expected owner.  No signature required";
```

Comment:
I would remove the "No signature required" part already as this will become the default soon.